classes/user_manager.php

```diff
@@ -0,0 +1,292 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * The class responsible for managing user data.
+ *
+ * @package     tool_userupsert
+ * @copyright   2021 Catalyst IT
+ * @license     https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_userupsert;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * The class responsible for managing user data.
+ *
+ * @package     tool_userupsert
+ * @copyright   2021 Catalyst IT
+ * @license     https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class user_manager {
+
+    /**
+     * A list of valid user statuses.
+     */
+    const VALID_USER_STATUSES = ['active', 'suspended', 'deleted'];
```

Comment:
Just a note that it was a bit confusing for me. Finally, I figured out that it's not a current or past user state. It's a desirable state that is going to be set as part of this userupsert call. Personally, it would be more clear for me if it's named like an _action_ or _command_. So you send a command to activate a user, or to delete a user. Feel free to let me know your thoughts

Reply:
this is from requirements. They want to send a status to be able to control user status in Moodle